log4j-core/src/main/java/org/apache/logging/log4j/core/layout/Rfc5424Layout.java

```diff
@@ -610,8 +610,8 @@ public static Rfc5424Layout createLayout(
             // @formatter:off
             @PluginAttribute(value = "facility", defaultString = "LOCAL0") final Facility facility,
             @PluginAttribute("id") final String id,
-            @PluginAttribute(value = "enterpriseNumber", defaultInt = DEFAULT_ENTERPRISE_NUMBER)
-            final int enterpriseNumber,
+            @PluginAttribute(value = "enterpriseNumber", defaultString = DEFAULT_ENTERPRISE_NUMBER)
+            final String enterpriseNumber,
```

Comment:
I think it is about time to deprecate this method and create a `Rfc5424Layout.Builder` class. The builder should validate the `enterpriseNumber` parameter in its `build()` method.

The signature of the method should remain the same.

Reply:
Deprecating this constructor and introducing a Builder feels like it could be a separate MR? In case I still should give it a shot, what do you mean by "validate" the parameter? What kind o f validation would you expect there?

Reply:
Note that when implementing the builder the constructor it calls should be private.